pkg/controllers/raycluster_webhook.go

```diff
@@ -77,6 +81,30 @@ func (w *rayClusterWebhook) Default(ctx context.Context, obj runtime.Object) err
 		rayCluster.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName = rayCluster.Name + "-oauth-proxy"
 	}
 
+	// add default queue label if not present
+	if rayCluster.GetLabels() == nil {
+		rayCluster.Labels = make(map[string]string)
+	}
+	err := w.Client.List(ctx, &kueuev1beta1.LocalQueueList{})
+	if err != nil {
+		rayclusterlog.Error(err, "Failed to list LocalQueues, Kueue CRD might not be installed")
+		_, ok := rayCluster.Labels["kueue.x-k8s.io/queue-name"]
+		if !ok {
+			// check if CRD Kueue LocalQueue exists
+			localQueues := &kueuev1beta1.LocalQueueList{}
```

Comment:
Could `metav1.PartialObjectMetadata` be used there, so it uses the partial metadata client / cache?